docs/pages/api/access-plugin.mdx

```diff
@@ -0,0 +1,761 @@
+  to complete this guide, though Go knowledge is required if you want to build
+  your own Access Request plugin.
+
+You will need the following in order to set up the demo plugin, which requires
+authenticating to the Google Sheets API:
+
+- A Google Cloud project with permissions to create service accounts.
+- A Google account that you will use to create a Google Sheets spreadsheet. We
+  will grant permissions to edit the spreadsheet to the service account used for
+  the plugin.
+
+<Admonition type="tip">
+
+Even if you do not plan to set up the demo project, you can follow this guide to see
+which libraries, types, and functions you can use to develop an Access Request
+plugin. 
+
+The demo is a minimal working example, and you can see fully fledged plugins in
+the
+[`gravitational/teleport-plugins`](https://github.com/gravitational/teleport-plugins)
+repository on GitHub.
+
+</Admonition>
+
+## Step 1/7. Set up your Go project
+
+The first step is to create a project directory and initialize a Go module for
+your project: 
+
+```code
+$ mkdir teleport-sheets
+$ cd teleport-sheets
+$ go mod init teleport-sheets
+go: creating new go.mod: module teleport-sheets
+```
+
+This creates a new file in your project directory, `go.mod`, which Go's package
+management system uses to fetch your plugin's dependencies.
+
+## Step 2/7. Set up the Google Sheets API
+
+Access Request plugins typically communicate with two APIs. They receive Access
+Request events from the Teleport Auth Service's gRPC API, and use the data to
+interact with the API of your chosen messaging or collaboration tool.
+
+In this section, we will enable the Google Sheets API, create a Google Cloud
+service account for the plugin, and use the service account to authenticate the
+plugin to Google Sheets.
+
+### Enable the Google Sheets API
+
+Enable the Google Sheets API by visiting the following Google Cloud console URL:
+
+https://console.cloud.google.com/apis/enableflow?apiid=sheets.googleapis.com
+
+Ensure that your Google Cloud project is the one you intend to use. 
+
+Click **Next** > **Enable**.
+
+### Create a Google Cloud service account for the plugin
+
+Visit the following Google Cloud console URL:
+
+https://console.cloud.google.com/iam-admin/serviceaccounts
+
+Click **Create Service Account**.
+
+For **Service account name**, enter "Teleport Google Sheets Plugin". Google
+Cloud will populate the **Service account ID** field for you.
+
+Click **Create and Continue**. When prompted to grant roles to the service
+account, click **Continue** again. We will create our service account without
+roles. Skip the step to grant users access to the service account, clicking
+**Done**.
+
+The console will take you to the **Service accounts** view. Click the name of
+the service account you just created, then click the **Keys** tab. Click **Add
+Key**, then **Create new key**. Leave the **Key type** as "JSON" and click
+**Create**.
```

Comment:
I suggest telling the reader to save the file as `credentials.json`, since that filename is expected in the code below.